tools/print-change-list.py

```diff
@@ -0,0 +1,194 @@
+        if pr.get('merged_at'):
+            return dateutil.parser.parse(pr['merged_at'])
+        else:
+            return None
+
+    # The GitHub API does not provide a `since` parameter to retrieve PRs
+    # closed since a given date, so instead we iterate over PRs in descending
+    # order of last update and stop when we reach a PR that was last updated
+    # before the given tag was created.
+    for closed_pr in github_request(auth_token, repo, 'pulls', state='closed',
+                                    sort='updated', direction='desc'):
+        pr_date = dateutil.parser.parse(closed_pr['updated_at'])
+        if pr_date < tag_date:
+            break
+        merged_at = merge_date(closed_pr)
+        if merged_at and merged_at > tag_date:
+            prs += [closed_pr]
+
+    return sorted(prs, key=merge_date)
+
+
+def format_list(items):
+    def format_item(item, col_width):
+        formatted = ''
+        for line in iter(textwrap.wrap(item, col_width - 2)):
+            if len(formatted) == 0:
+                formatted = formatted + '- ' + line
+            else:
+                formatted = formatted + '\n  ' + line
+        return formatted
+
+    return '\n\n'.join([format_item(item, 80) for item in items])
+
+
+def current_github_repo(remote='origin'):
+    """
+    Return the GitHub repository and organization associated with a Git remote.
+
+    This function assumes that `remote` has a SSH or HTTPS URL containing
+    'github.com' and ending with '{org}/{repo}.git'.
+    """
+
+    proc = run(['git', 'remote', 'get-url', remote], stdout=PIPE)
+    url = proc.stdout.strip().decode('utf-8')
+
+    if 'github.com' in url:
+        match = re.search('([^:/]+)/([^/]+)\.git$', url)
+        org = match.group(1)
+        repo = match.group(2)
+        return (org, repo)
+    else:
+        return None
+
+    
+def main():
+    parser = argparse.ArgumentParser(description=
+"""
+Generates a list of changes since the last tag was created, in the format
+recommended by http://keepachangelog.com.
+"""
+)
+
+    current_repo = ''
+    try:
+        (org, repo) = current_github_repo()
+        current_repo = '{}/{}'.format(org, repo)
+    except TypeError:
+        pass
+
+    parser.add_argument('--tag', default=get_last_tag(), help='The tag to list changes since')
+    parser.add_argument('--repo', default=current_repo, help='The GitHub repository to generate a change list for')
+    parser.add_argument('--token', default=getenv('GITHUB_TOKEN'), help='GitHub API access token')
+    args = parser.parse_args()
+
+    if not args.repo:
+        print('Could not determine current GitHub repo. Use the --repo argument',file=stderr)
+        exit(1)
+
+    if not args.tag:
+        print('Could not find last tag. Use the --tag argument', file=stderr)
+        exit(1)
+
+    pr_details = []
+    for pr in get_prs_merged_since(args.token, args.repo, args.tag):
+        pr_details += [PRInfo(pr['number'], title=pr['title'].strip())]
```

Comment:
What do you think of making this a little more object oriented? I think it might make the code read a little better. I'm thinking of something along the lines of renaming the `PRInfo` class to `PullRequest` and having `get_prs_merged_since()` be a `PullRequest` class method and return a list of `PullRequest` objects. Then these lines of code could look something like:

```python
pull_requests = PullRequest.merged_since(args.token, args.repo, args.tag)
```

If you wanted to take it even further you could have `GitHubRepo` and `GitHub` classes to wrap up more of the arguments:

```python
github = GitHub(args.token)
repo = GitHubRepo(github, args.repo)
pull_requests = repo.prs_merged_since(args.tag)
```